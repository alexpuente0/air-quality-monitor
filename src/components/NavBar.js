import { useNavigate } from 'react-router-dom';
import {
  IoColorPaletteSharp,
  IoCloseSharp,
  IoChevronBackSharp,
} from 'react-icons/io5';
import { useState } from 'react';
import './styles/navBar.scss';

const Navbar = () => {
  const navigate = useNavigate();

  const [modalVisible, setModalVisible] = useState(false);
  const [theme, setTheme] = useState({
    color: ['Pink', 'Blue'],
    isPink: true,
  });

  const handleTheme = () => {
    if (theme.isPink) {
      document.documentElement.style.setProperty('--primary-color', '#2a4476');
      document.documentElement.style.setProperty(
        '--secondary-color',
        '#2a4476',
      );
      document.documentElement.style.setProperty('--tertiary-color', '#3d60a4');
      document.documentElement.style.setProperty(
        '--quaternary-color',
        '#3d60a4',
      );
    } else {
      document.documentElement.style.setProperty('--primary-color', '#7a183e');
      document.documentElement.style.setProperty(
        '--secondary-color',
        '#ad0247',
      );
      document.documentElement.style.setProperty('--tertiary-color', '#ac144e');
      document.documentElement.style.setProperty(
        '--quaternary-color',
        '#ac144e',
      );
    }
    setTheme({ ...theme, isPink: !theme.isPink });
  };

  return (
    <nav className="nav-bar">
      <ul className="nav-links">
        <li>
          <IoChevronBackSharp
            className="nav-icon"
            onClick={() => navigate(-1)}
          />

        </li>
        <li className="nav-item">Air Quality World Monitor</li>
        <li>
          <IoColorPaletteSharp
            data-testid="icon"
            className="nav-icon"
            onClick={() => setModalVisible(!modalVisible)}
          />
        </li>
      </ul>
      {modalVisible && (
        <div className="modal">
          <div className="modal-content">
            <button className="theme" onClick={handleTheme} type="button">
              Change Theme:
              {theme.isPink ? theme.color[1] : theme.color[0]}
            </button>
            <IoCloseSharp
              className="close"
              onClick={() => setModalVisible(!modalVisible)}
            />
          </div>
        </div>
      )}
    </nav>
  );
};

export default Navbar;
