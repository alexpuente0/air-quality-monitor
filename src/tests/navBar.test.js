import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NavBar from '../components/navBar';

const MockedNavbar = () => (
  <BrowserRouter>
    <NavBar />
  </BrowserRouter>
);

describe('Navbar component', () => {
  it('renders succesfully', () => {
    render(<MockedNavbar />);
    const listedItem = screen.getByText('Air Quality World Monitor');
    expect(listedItem).toMatchSnapshot();
  });

  it('change theme', () => {
    render(<MockedNavbar />);
    const listedItem = screen.getByTestId('icon');
    fireEvent.click(listedItem);
    const button = screen.getByRole('button', 'Change Theme:Blue');
    fireEvent.click(button);
    expect(button.innerHTML).toBe('Change Theme:Pink');
  });
});
