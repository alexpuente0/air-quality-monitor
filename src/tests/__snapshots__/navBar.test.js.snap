// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component renders succesfully 1`] = `
<li
  class="nav-item"
>
  Air Quality World Monitor
</li>
`;
